Tests/MongoSwiftTests/CommandMonitoringTests.swift

```diff
@@ -104,71 +97,72 @@ final class CommandMonitoringTests: XCTestCase {
     }
 }
 
 /// A struct to hold the data for a single test from a CMTestFile.
-private struct CMTest {
+private struct CMTest: Decodable {
+
+    struct Operation: Decodable {
+        let name: String
+        let args: Document
+        let readPreference: Document?
+
+        enum CodingKeys: String, CodingKey {
+            case name, args = "arguments", readPreference
+        }
+    }
+
+    let op: Operation
     let description: String
-    let operationName: String
-    let args: Document
-    let readPreference: Document?
-    let expectations: [ExpectationType]
+    let expectationDocs: [Document]
     let minServerVersion: String?
     let maxServerVersion: String?
 
     // Some tests contain cursors/getMores and we need to verify that the
     // IDs are consistent across sinlge operations. we store that data in this
     // `context` dictionary so we can access it in future events for the same test
-    var context: [String: Any]
-
-    init(fromDocument document: Document) throws {
-        self.description = try document.get("description")
-        let operation: Document = try document.get("operation")
-        self.operationName = try operation.get("name")
-        self.args = try operation.get("arguments")
-        self.readPreference = operation["readPreference"] as? Document
-        let expectationDocs: [Document] = try document.get("expectations")
-        self.expectations = try expectationDocs.map { try makeExpectation($0) }
-        self.minServerVersion = document["ignore_if_server_version_less_than"] as? String
-        self.maxServerVersion = document["ignore_if_server_version_greater_than"] as? String
-        self.context = [String: Any]()
+    var context = [String: Any]()
+
+    var expectations: [ExpectationType] { return try! expectationDocs.map { try makeExpectation($0) } }
+
+    enum CodingKeys: String, CodingKey {
+        case description, op = "operation", expectationDocs = "expectations",
+        minServerVersion = "ignore_if_server_version_less_than",
+        maxServerVersion = "ignore_if_server_version_greater_than"
     }
 
     // Given a collection, perform the operation specified for this test on it.
-    // Wrap each operation in do/catch because we expect some of them to fail.
+    // try? each operation because we expect some of them to fail.
     // If something fails/succeeds incorrectly, we'll know because the generated
     // events won't match up.
     func doOperation(withCollection collection: MongoCollection<Document>) throws {
         // TODO SWIFT-31: use readPreferences for commands if provided
-        let filter = self.args["filter"] as? Document
-        switch self.operationName {
+        let filter: Document = self.op.args["filter"] as? Document ?? [:]
+
+        switch self.op.name {
 
         case "count":
-            do { _ = try collection.count(filter ?? [:]) } catch { }
+            _ = try? collection.count(filter)
```

Comment:
Is the `_` assignment necessary just so we can use `try?` and avoid a verbose `try`/`catch` block?

Reply:
yup exactly

Reply:
What happens with this omitted? Is there a compile error saying that we're discarding the return value?